pkg/config/validator.go

```diff
@@ -72,19 +73,35 @@ type (
 	// AdapterToAspectMapper returns the set of aspect kinds implemented by
 	// the given builder.
 	AdapterToAspectMapper func(builder string) KindSet
+
+	// BuilderInfoFinder is used to find specific handlers BuilderInfo for configuration.
+	BuilderInfoFinder func(name string) (*adapter.BuilderInfo, bool)
+
+	// SetupHandlerFn is used to configure handler implementation with Types associated with all the templates that
+	// it supports.
+	SetupHandlerFn func(actions []*pb.Action, constructors map[string]*pb.Constructor,
```

Comment:
Sorry, didn't intend for you to do anything as a result of this. More a note to myself alongside the refactor/cleanup I suggested (which I also don't intend for you to have to do :) ).

I meant that in the original version, there was a `ValidatorInit` struct (making up the name, it was called something else but similar) whose fields were the values that make up the `newValidator` func args. So `func newValidator(init *ValidatorInit) *validator { ... }`. At the time I pushed back against that pattern, but the args list is getting ridiculous.

I'm planning on refactoring this whole file, I'll take a whack at the args while I'm at it.